### ISSUE TITLE
Heap settings not applied for Cassandra 4.0 images

### ISSUE THREAD

**Issue by emerkle826** [k8ssandra/k8ssandra#394]
## Bug Report

**Describe the bug**
For Cassandra 4.0 images, K8ssandra heap settings are not honored.

**To Reproduce**
Specify heap settings in K8ssandra values similar to this
```yaml
cassandra:
  version: "4.0.0"
  auth:
    enabled: false
  clusterName: k8ssandra
  heap:
    size: "2G"
    newGenSize: "256M"

```
On my laptop, with or without the heap specified, I see the Cassandra process started with almost 8GB of min/max heap specified:

```
... -Xms7990M -Xmx7990M -XX:CompileCommandFile=/opt/cassandra/conf/hotspot_compiler ...
```

This is because k8ssandra puts the heap settings in `jvm.options`. However, that file isn't used for Cassandra 4.0 (only for 3.x). Cassandra 4.0 looks for server options in `jvm-server.options` (for all running JVMs) and then `jvm8-server.options` or `jvm11-server.options`, depending on the running JRE version.

**Expected behavior**
The Cassandra process should run with the specified heap settings.

**Environment (please complete the following information):**

* Helm charts version info
```helm ls -A
NAME     	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART           	APP VERSION
k8ssandra	default  	1       	2021-02-19 14:45:34.221944528 -0600 CST	deployed	k8ssandra-0.53.0	3.11.10
```

* Helm charts user-supplied values
N/A

* Kubernetes version information:
N/A

* Kubernetes cluster kind:
N/A

**Comment by emerkle826** [comment]
This should be resolved by #336

**Comment by jdonenine** [comment]
Resolved in #336
